### PR TITLE
Fix no such type error when anonymous error types in imported modules are used

### LINKED ISSUE #39483: [Task]: Get anonymous error types in imported modules from `getAnonType` method
### Description

$title

### Describe your task(s)

_No response_

### Related area

-> Runtime

### Related issue(s) (optional)

_No response_

### Suggested label(s) (optional)

_No response_

### Suggested assignee(s) (optional)

_No response_

### PR DESCRIPTION
**Pull request by Nadeeshan96**
$title
Fixes #39483

## Check List 
- [x] Read the [Contributing Guide](https://github.com/ballerina-platform/ballerina-lang/blob/master/CONTRIBUTING.md)
- [ ] Updated Change Log
- [ ] Checked Tooling Support (#<Issue Number>)
- [x] Added necessary tests
  - [x] Unit Tests
  - [ ] Spec Conformance Tests
  - [ ] Integration Tests
  - [ ] Ballerina By Example Tests
- [ ] Increased Test Coverage   
- [ ] Added necessary documentation  
  - [ ] API documentation 
  - [ ] Module documentation in Module.md files
  - [ ] Ballerina By Examples

### CODE REVIEW

File: .github/workflows/pull_request_full_build.yml [hunk 1]
```diff
@@ -4,6 +4,7 @@ on:
   pull_request:
     branches:
       - master
+      - 2201.[4-9]+.[0-9]+-stage
```
Comment (gabilang) [inline .github/workflows/pull_request_full_build.yml:7]: Do we need this change since the fix being added to the `master` branch?

Comment (Nadeeshan96) [reply .github/workflows/pull_request_full_build.yml:7]: I will remove this.

File: compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/TypeHashVisitor.java [hunk 2]
```diff
@@ -563,19 +555,11 @@ private boolean isCyclic(BType type) {
...
 
     private Integer addToVisited(BType type, Integer hash) {
-        Integer existing = Optional.ofNullable(generated.get(hash)).orElse(0);
-
-        generated.put(hash, existing + 1);
-        if (existing > 0 && !visited.containsKey(type)) {
-            hash += existing;
-        }
-        assert visiting.pop() == type;
         visited.put(type, hash);
```
Comment (warunalakshitha) [inline compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/TypeHashVisitor.java:563]: should we have early return if type is already in visited?

Comment (Nadeeshan96) [reply compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/TypeHashVisitor.java:563]: Early return is there before this method is called. Created https://github.com/ballerina-platform/ballerina-lang/issues/39689 to remove duplicate code.

**Review by warunalakshitha — APPROVED** [review]
LGTM

**Pull request merged**